ark/schema/shared/traversal.ts

```diff
@@ -149,8 +149,11 @@ export class TraversalContext {
 	pathHasError(path: TraversalPath): boolean {
 		if (!this.hasError()) return false
 
-		const propString = pathToPropString(path)
-		return this.errors.some(e => propString.startsWith(e.propString))
+		for (let i = 0; i <= path.length; i++) {
+			const partialPropString = pathToPropString(path.slice(0, i))
```

Comment:
`pathToPropString` already has to iterate the path. Look at that logic and reuse the internal methods so build it here as you go so this isn't n^2